src/app/admin/components/sidebar/sidebar.component.ts

```diff
@@ -1,25 +1,54 @@
-import { Component, OnInit } from '@angular/core';
-import { Location } from '@angular/common';
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
+import { Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { Collection } from 'app/core/collection.service';
+import { trigger, query, style, animate, transition } from '@angular/animations';
 
 @Component({
   selector: 'clark-admin-sidebar',
   templateUrl: './sidebar.component.html',
-  styleUrls: ['./sidebar.component.scss']
+  styleUrls: ['./sidebar.component.scss'],
+  animations: [
+    trigger('sidebar', [
+      transition('* => *', [
+        query(':enter', [
+          style({ opacity: 0, height: 0 }),
+          animate('200ms ease', style({ opacity: 1, height: '*' }))
+        ], { optional: true }),
+        query(':leave', [
+          style({ opacity: 1, height: '*' }),
+          animate('200ms ease', style({ opacity: 0, height: 0 }))
+        ], { optional: true })
+      ])
+    ])
+  ]
 })
-export class SidebarComponent implements OnInit {
+export class SidebarComponent implements OnInit, OnDestroy {
+  destroyed$: Subject<void> = new Subject();
 
-  constructor(private location: Location) { }
+  @Input() collections: string[] = [];
+  @Input() activeCollection: string;
+  @Input() editorMode: boolean;
 
-  ngOnInit() {
-  }
+  @Input() initialized = false;
+
+  constructor(private router: Router) {}
+
+  ngOnInit() {}
```

Comment:
If we aren't using this lifecycle hook we should remove it and the implementation of the interface.